testplan/runners/pools/tasks/base.py

```diff
@@ -231,43 +231,44 @@ def materialize(self, target=None):
 
     def _string_to_target(self):
         """Dynamically load an object from a module by target name."""
-        path_inserted = False
-        if isinstance(self._path, str):
-            sys.path.insert(0, self._path)
-            path_inserted = True
+        sys.path.insert(0, self._path)
+        module_imported = False
 
         try:
             if self._module is None:
-                ###################################################
-                # For backward compatibility, will be removed later
-                elements = self._target.split(".")
-                target_src = elements.pop(-1)
-                if len(elements):
-                    mod = importlib.import_module(".".join(elements))
-                    warnings.warn(
-                        "DEPRECATED: "
-                        "If a test target is scheduled in string format, the "
-                        "module from where it is imported should be specified"
+                try:
+                    module, target = self._target.rsplit(".", 1)
+                except ValueError:
+                    raise TaskMaterializationError(
+                        "Task parameters are not sufficient for"
+                        " target {} materialization".format(self._target)
                     )
-                    return getattr(mod, target_src)
-                ####################################################
-                raise TaskMaterializationError(
-                    "Task parameters are not sufficient for"
-                    " target {} materialization".format(self._target)
-                )
-            parent_target, target = importlib.import_module(self._module), None
-            for element in self._target.split("."):
-                target = getattr(parent_target, element, None)
-                if target is None:
+            else:
+                module = self._module
+                target = self._target
+
+            if module in sys.modules:
+                mod = sys.modules[module]
+                module_imported = False
+            else:
+                mod = importlib.import_module(module)
+                module_imported = True
```

Comment:
This looks to be a pattern used elsewhere does it worth a utility function?

Maybe a context manager that auto unload the module when imported from here?

Reply:
good suggestion, i have revised this part.